test/src/loaders/ncbi_downloader/ncbi_downloader_test.py

```diff
@@ -1,5 +1,103 @@
+import os
+import shutil
+import uuid
+from pathlib import Path
+
+import pytest
+
 from src.loaders.ncbi_downloader import ncbi_downloader
+from src.loaders.ncbi_downloader.ncbi_downloader import GTDB_DOMAIN
+
+
+@pytest.fixture(scope="module")
+def setup_and_teardown():
+    print('starting NCBI downloader tests')
+    tmp_dir = 'result_{}'.format(str(uuid.uuid4()))
+    os.makedirs(tmp_dir, exist_ok=True)
+
+    caller_filename_full = Path(__file__).resolve()
+    caller_file_dir = os.path.dirname(caller_filename_full)
+
+    project_dir = Path(caller_file_dir).resolve().parents[3]
+    script_file = f'{project_dir}/src/loaders/ncbi_downloader/ncbi_downloader.py'
+
+    yield tmp_dir, script_file
+
+    shutil.rmtree(tmp_dir, ignore_errors=True)
+
+
+def test_make_work_dir(setup_and_teardown):
+    tmp_dir, script_file = setup_and_teardown
+
+    with pytest.raises(ValueError, match='Unexpected source:'):
+        fake_source = 'hello_fake'
+        _ = ncbi_downloader._make_work_dir(tmp_dir, fake_source, 'release_ver')
```

Comment:
why `=`

Reply:
removed. I did it sometimes just to remind me the method actually returns something.

Reply:
:+1: